Skip preserving file extension for generated path

Babel would emit the files with js extension irrespective of the extension of source files, which would cause problem to test individual files without using webpack or gulp,  since required modules path would never be resolved, becuase generated files would have an extension of `js` and the generated webpack alias path would be referring to `jsx`(i.e original extension)

I've shown an example to support the use case

### .babelrc
```js
{
  "presets": ["es2015", "react","stage-2"],
  "plugins" : [
  ["babel-plugin-webpack-alias", { "config": "./webpack/server/server.js" }]
  ],
}
```

### webpack config
```js
resolve : {
    extensions: ['','.js','.jsx'],
    modules : [srcPath,path.join(basePath,'./node_modules')],
    alias : {
      template: path.join(srcPath,'./template')
    },
  },
```

### es6 code
```js
import Layout from 'template/Layout';
```

### Generating es5 file post babel transformation
```console
src/server/template/Layout.jsx -> lib/server/template/Layout.js
```

### Post babel transformation 
```js
var _Layout = require('./Layout.jsx'); //should be './Layout.js'
var _Layout2 = _interopRequireDefault(_Layout);
```

### Running in node
```console
Error: Cannot find module './Layout.jsx'
```

--- a/src/index.js
+++ b/src/index.js
@@ -47,7 +47,7 @@ function getConfigPath(configPaths, findConfig) {
 export default function({ types: t }) {
     return {
         visitor: {
-            CallExpression(path, { file: { opts: { filename } }, opts: { config: configPath, findConfig: findConfig = false } = {} }) {
+            CallExpression(path, {file: { opts: { filename } }, opts: { config: configPath, findConfig: findConfig = false, noOutputExtension = false } = {} }) {
                 const configPaths = configPath ? [configPath, ...DEFAULT_CONFIG_NAMES] : DEFAULT_CONFIG_NAMES;
 
                 // Get webpack config
@@ -182,7 +182,7 @@ export default function({ types: t }) {
                             }
 
                             // In case the extension option is passed
-                            if(extensionsConf) {
+                            if(extensionsConf && !noOutputExtension) {
                                 // Get an absolute path to the file
                                 const absoluteRequire = join(aliasTo, basename(filePath));
 

--- a/test/fixtures/no-extension/expected.js
+++ b/test/fixtures/no-extension/expected.js
@@ -1,0 +1,6 @@
+'use strict';
+
+require('../required/myfile');
+require('../required/doesntExist');
+
+// Rest of the file

--- a/test/fixtures/no-extension/source.js
+++ b/test/fixtures/no-extension/source.js
@@ -1,0 +1,4 @@
+require('my-absolute-test-lib/myfile');
+require('my-absolute-test-lib/doesntExist');
+
+// Rest of the file

--- a/test/index.spec.js
+++ b/test/index.spec.js
@@ -106,3 +106,8 @@ test('works with webpack configs that export an array, instead of a single objec
     t.is(actual, expected);
 });
 
+test('doesnt output extensions when noOutputExtension is set to true', t => {
+    const actual = transformFixture('no-extension/source.js', {config: './extensions.config.js', noOutputExtension: true});
+    const expected = readFixture('no-extension/expected.js');
+    t.is(actual, expected);
+});

